bigc/resources/checkouts.py

```diff
@@ -0,0 +1,92 @@
+from typing import Union
+from urllib.parse import urlencode, urlparse, urlunparse
+from uuid import UUID
+
+from bigc.api_client import BigCommerceAPIClient
+
+UUIDLike = Union[UUID, str]
+
+
+class BigCommerceCheckoutsAPI:
+    def __init__(self, api_client: BigCommerceAPIClient):
+        self._api = api_client
+
+    def get(self, checkout_id: UUIDLike) -> dict:
+        """Get a specific checkout by its ID"""
+        return self._api.v3.get(f'/checkouts/{checkout_id}')
+
+    def update_customer_message(self, checkout_id: UUIDLike, *, customer_message: str) -> dict:
+        """Change customer message pertaining to an existing Checkout"""
+        payload = {'customer_message': customer_message}
+        return self._api.v3.put(f'/checkouts/{checkout_id}', json=payload)
+
+    def add_billing_address(self, checkout_id: UUIDLike, *, email: str, country_code: str, **kwargs) -> dict:
+        """Add a billing address to an existing checkout"""
+        payload = {'email': email, 'country_code': country_code, **kwargs}
+        return self._api.v3.post(f'/checkouts/{checkout_id}/billing-address', json=payload)
+
+    def update_billing_address(self, checkout_id: UUIDLike, address_id: str, *, email: str, country_code: str,
+                               **kwargs) -> dict:
+        """Update an existing billing address on a checkout"""
+        payload = {'email': email, 'country_code': country_code, **kwargs}
+        return self._api.v3.put(f'/checkouts/{checkout_id}/billing-address/{address_id}', json=payload)
+
+    def add_consignment(self, checkout_id: UUIDLike,
+                        include_available_shipping_options: bool = False,
+                        **kwargs) -> dict:
+        """Add a new consignment to a checkout"""
+        url_parts = urlparse(f'/checkouts/{checkout_id}/consignments')
+
+        include = []
+        if include_available_shipping_options:
+            include.append('consignments.available_shipping_options')
+
+        query_dict = {}
+        if include:
+            query_dict['include'] = ','.join(include)
+
+        url_parts = url_parts._replace(query=urlencode(query_dict))
+
+        payload = [{**kwargs}]
+        return self._api.v3.post(urlunparse(url_parts), json=payload)
+
+    def update_consignment_shipping_option(self, checkout_id: UUIDLike, consignment_id: str,
+                                           include_available_shipping_options: bool = False,
+                                           **kwargs) -> dict:
```

Comment:
I think this needs to be renamed to just `update_consignment` now since we now support updating more than just the selected shipping option.